src/Security/Authentication/FrontendPreviewAuthenticator.php

```diff
@@ -94,17 +100,37 @@ public function authenticateFrontendUser($username = null): void
                 );
             }
 
-            return;
+            return false;
         }
 
-        $token = new UsernamePasswordToken($user, null, $providerKey, (array) $user->getRoles());
+        $allowedGroups = \StringUtil::deserialize($backendUser->amg, true);
+        $frontendGroups = \StringUtil::deserialize($frontendUser->groups, true);
```

Comment:
CS

Reply:
what CS?

Reply:
`\StringUtil` in new code.  
Shouldn't it be `Use Contao\StringUtil;` and then `StringUtil:: deserialize` ?

Reply:
Don't worry about those things, I will fix them all before merging. Focus on the "real" issues. 😄